### PR TITLE
PaginationIterator storing only one page at a time

### PR DESCRIPTION
**Pull request by ecoologic**
I'd rather tell users to not use `iterator_to_array` than to store all pages in memory and likely cause bigger issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,7 +155,10 @@ foreach ($iterator as $ticket) {
   * Refer to the docs for details, including allowed sort fields
 * Combine everything: `$params = ['page[size]' => 2, 'sort' => 'updated_at', 'extra' => 'param'];`
 
-**Note**: Refer to the documentation for the correct params for sorting with the pagination type you're using.
+**Note**:
+
+* Refer to the documentation for the correct params for sorting with the pagination type you're using
+* The helper method `iterator_to_array` doesn't work with this implementation
```
Comment (ecoologic) [inline README.md:161]: I'd love to find out why, but I don't have time and it's hard for me to dig into that.

File: src/Zendesk/API/Traits/Utility/Pagination/ObpStrategy.php [hunk 2]
```diff
@@ -18,7 +18,7 @@ public function page($getPageFn)
         return $response->{$this->resourcesKey};
     }
 
-    public function shouldGetPage($position) {
-        return $this->pageNumber == 0 || $position >= $this->pageNumber * $this->pageSize();
+    public function shouldGetPage($current_page) {
+        return $this->pageNumber == 0 || count($current_page) == 0;
```
Comment (ecoologic) [inline src/Zendesk/API/Traits/Utility/Pagination/ObpStrategy.php:22]: This is unrelated to the single page fetch, which in fact affects CBP as well.

**Review by fbvilela — APPROVED** [review]
